chanfitness/chanevent.go

```diff
@@ -128,51 +134,72 @@ func (e *chanEventLog) getOnlinePeriods() []*onlinePeriod {
 	// recent event is tracked using the offline bool so that we can add a
 	// final online period if necessary.
 	for _, event := range e.events {
-
 		switch event.eventType {
 		case peerOnlineEvent:
-			lastOnline = event.timestamp
-			offline = false
+			// If our previous event is nil, we just set it and
+			// break out of the switch.
+			if lastEvent == nil {
```

Comment:
I’ve had another look at this, and I think that we may be ok in terms of ordering because we notify on quite a high level. We notify peer online events in `addPeer` and offline events in `removePeer`. 

`addPeer` is only called when we get an inbound connection that we do not have an existing record of, or we replace a stale connection with a fresh one (`InboundPeerConnected`/ `OutboundPeerConnected`). When we are replacing stale connections, we remove all previous state, which will trigger an offline event before our new online event. 

This happens at a much higher level than our connection retries, and has logic in place that’s aware of existing connections. This does make our online events less granular, but it happens at the same level for every peer (and is preexisting for this uptime measure).  

However, it’s still difficult to say for sure whether we can rely on in-order events in reality. Perhaps @Crypt-iQ  can weigh in on this? I still like the suggestion of counting online and offline events, because it makes sure that our measurement isn’t biased (perhaps certain peers have certain odd connection habits, others don’t). Esp in this case, where the value itself matters less than the comparison of different peers values.